Assets/FPS/Scripts/Gameplay/Nautica/ChallengeLevelManager.cs

```diff
@@ -31,5 +48,18 @@ public override void CheckForEndOfEpisodeEvent()
 				return;
 			}
 		}
+
+		protected override void MoveToNextLevel()
+		{
+			ResetAgentHealth();
+			challengeManager.SetUpNextLevel();
+		}
+
+		private void ResetAgentHealth()
+        {
+			var agentHealth = agentObj.GetComponent<Health>();
+			agentHealth.Revive();
+			challengeManager.ResetHealthBar();
```

Comment:
not asking to change this, but just wanted to offer up some info you can look into -- you can use events to handle some of these types of things.
There are UnityEvent and System.Action.  In this case you could have `agentHealth.Revive()` code triggers an event like "reviveEvent".  then the health bar listens for this event and when it happens, it executes code that resets the healthbar.  We're not using these in any of our code now, and we dont want to go around modifying everything at the end of our project.  but just giving you stuff you can try for future projects